src/cascadia/TerminalApp/ColorScheme.cpp

```diff
@@ -120,36 +131,60 @@ void ColorScheme::LayerJson(const Json::Value& json)
     }
 }
 
-winrt::hstring ColorScheme::Name() const noexcept
-{
-    return _schemeName;
-}
-
-winrt::com_array<Color> ColorScheme::Table() const noexcept
+// Method Description:
+// - Create a new serialized JsonObject from an instance of this class
+// Arguments:
+// - scheme: a ColorScheme object that will be converted into a serialized JsonObject
+// Return Value:
+// - a serialized JsonObject from the values in scheme
+Json::Value ColorScheme::ToJson(const TerminalApp::ColorScheme& scheme)
 {
-    winrt::com_array<Color> result{ COLOR_TABLE_SIZE };
-    std::transform(_table.begin(), _table.end(), result.begin(), [](til::color c) -> Color { return c; });
-    return result;
+    Json::Value json{ Json::ValueType::objectValue };
+    const auto schemeImpl{ winrt::get_self<implementation::ColorScheme>(scheme) };
+    schemeImpl->UpdateJson(json);
+    return json;
 }
 
-Color ColorScheme::Foreground() const noexcept
+// Method Description:
+// - Update the given json object with values from this object.
+// Arguments:
+// - json: an object which will be a serialization of a ColorScheme object.
+// Return Value:
+// <none>
+void ColorScheme::UpdateJson(Json::Value& json)
 {
-    return _defaultForeground;
-}
+    JsonUtils::SetValueForKey(json, NameKey, _Name);
+    JsonUtils::SetValueForKey(json, ForegroundKey, _Foreground);
+    JsonUtils::SetValueForKey(json, BackgroundKey, _Background);
+    JsonUtils::SetValueForKey(json, SelectionBackgroundKey, _SelectionBackground);
+    JsonUtils::SetValueForKey(json, CursorColorKey, _CursorColor);
 
-Color ColorScheme::Background() const noexcept
-{
-    return _defaultBackground;
+    int i = 0;
+    for (const auto& current : TableColors)
+    {
+        JsonUtils::SetValueForKey(json, current, _table.at(i));
+        i++;
+    }
 }
 
-Color ColorScheme::SelectionBackground() const noexcept
+winrt::com_array<Color> ColorScheme::Table() const noexcept
 {
-    return _selectionBackground;
+    winrt::com_array<Color> result{ COLOR_TABLE_SIZE };

```

Comment:
might want to prefer basing it on `_table`'s size, just in case we ever make it so you can set the whole 256-color palette